assets/app/scripts/controllers/project.js

```diff
@@ -88,58 +67,59 @@ angular.module('openshiftConsole')
         var servicePods = [];
         for (var selectorKey in service.selector) {
           var selectorValue = service.selector[selectorKey];
-          var pods = $scope.podsByLabel[selectorKey][selectorValue];
-          for (var podId in pods) {
-            var pod = pods[podId];
-            servicePods.push(pod);
+          if ($scope.podsByLabel[selectorKey]) {
+            var pods = $scope.podsByLabel[selectorKey][selectorValue];
+            for (var podId in pods) {
+              var pod = pods[podId];
+              servicePods.push(pod);
+            }
           }
         }
         $scope.podsByServiceByLabel[serviceId]  =  {};
+        // TODO last remaining reference to this... 
         DataService.objectsByAttribute(servicePods, "labels", $scope.podsByServiceByLabel[serviceId], null, "id");
       }
 
       console.log("podsByServiceByLabel", $scope.podsByServiceByLabel);      
     };
 
-    $scope.podsPromise.done($.proxy(function(pods){
-      $scope.servicesPromise.done($.proxy(function(services) {
-        //we've loaded the initial set of pods and services at this point
-        // TODO we'll restructure how we handle this data to prevent things from jumping around
-        // in crazy structural cases by service
-        servicesCallback(services);
-        podsCallback(pods);
-
-        DataService.subscribe("services", servicesSubscribeCallback, $scope);
-        DataService.subscribePolling("pods", podsCallback, $scope);
-      }, this));
-    }, this));
-
     // Sets up subscription for deployments and deploymentsByConfig
-    var deploymentsCallback = function(action, deployment) {
+    var deploymentsCallback = function(deployments, action, deployment) {
       $scope.$apply(function() {
-        if (deployment.annotations && deployment.annotations.encodedDeploymentConfig) {
-          var depConfig = $.parseJSON(deployment.annotations.encodedDeploymentConfig);
-          deployment.details = depConfig.details;
+        $scope.deployments = deployments.by("id");
+        $scope.deploymentsByConfig = deployments.by("annotations.deploymentConfig", "id");
+        if (deployment) {
```

Comment:
1. Split the parsing of the encodedDeploymentConfig into a small method you can call from both places
```
function parseDeploymentConfig(dep, overwrite) {
  if (dep.details && !overwrite) return;
  if (dep.annotations && dep.annotations.encodedDeploymentConfig) {
    var depConfig = $.parseJSON(dep.annotations.encodedDeploymentConfig);
    dep.details = depConfig.details;
  }
}
```

2. Only process the individual deployment is the action is not "DELETED"?